phys2bids/bids_units.py

```diff
@@ -0,0 +1,94 @@
+
+unit_aliases = {
+                # ampere: electric current
+                'a': 'A', 'ampere': 'A', 'amp': 'A', 'amps': 'A',
+                # kelvin: thermodynamic temperature
+                'k': 'K', 'kelvin': 'K', 'kelvins': 'K',
+                # mole: amount of substance
+                'mol': 'mol', 'mole': 'mol',
+                # hertz: frequency
+                'hz': 'Hz', 'hertz': 'Hz',
+                '1/s': 'Hz', '1/second': 'Hz', '1/seconds': 'Hz',
+                '1/sec': 'Hz', '1/secs': 'Hz',
+                # newton: force, weight
+                'n': 'N', 'newton': 'N', 'newtons': 'N',
+                # pascal: pressure, stress
+                'pa': 'Pa', 'pascal': 'Pa', 'pascals': 'Pa',
+                # volt: voltage (electrical potential), emf
+                'v': 'V', 'volt': 'V', 'volts': 'V',
+                # degree Celsius: temperature relative to 273.15 K
+                '°c': '°C', 'celsius': '°C', '°celsius': '°C',
+                # second: time
+                'second': 's', 'seconds': 's', 'sec': 's',
+                'secs': 's',
+                '1/hz': 's', '1/hertz': 's', 's': 's'
+}
+
+# Init dictionary of aliases for multipliers. Entries are still lowercase
+prefix_aliases = {
+                    # Multiples - skip "mega" and only up to "tera"
+                    'da': 'da', 'deca': 'da', 'h': 'h', 'hecto': 'h',
+                    'k': 'k', 'kilo': 'k', 'g': 'G', 'giga': 'G', 't': 'T',
+                    'tera': 'T',
+                    # Submultipliers
+                    'd': 'd', 'deci': 'd', 'c': 'c', 'centi': 'c',
+                    'milli': 'm', 'm': 'm', 'µ': 'µ', 'micro': 'µ',
+                    'n': 'n', 'nano': 'n', 'p': 'p', 'pico': 'p',
+                    'f': 'f', 'femto': 'f', 'a': 'a', 'atto': 'a',
+                    'z': 'z', 'zepto': 'z', 'y': 'y', 'yocto': 'y',
+}
+
+
+def bidsify_units(orig_unit):
+    """
+    Read the input unit of measure and use the dictionary of aliases
+    to bidsify its value.
+    It is possible to make simple conversions
+
+    Parameters
+    ----------
+    unit: string
+        Unit of measure, might or might not be BIDS compliant.
+
+    Returns
+    -------
+    new_unit: str
+        BIDSified alias of input unit
+
+    Notes
+    -----
+    This function should implement a double check, one for unit and
+    the other for prefixes (e.g. "milli"). However, that is going to be tricky,
+    unless there is a weird way to multiply two dictionaries together.
+    """
+    # call prefix and unit dicts
+    new_unit = ""
+    # for every unit alias in the dict
+    for u_key in unit_aliases.keys():
+        unit = orig_unit.lower()
+        # check that u_key is part of unit
+        if u_key in unit:
+            new_unit = unit[-len(u_key):]
+            # check that it is at the end
+            if new_unit == u_key:
+                # leave only the prefix
+                unit = unit.replace(u_key, "")
```

Comment:
I think that `unit = unit[:len(u_key)]` might be safer here. For example, if someone (for whatever reason) was using units of nano Newtons and coded it as `nn` then this would convert the entire thing to `''`.

Reply:
Yeah that's true but ``unit = unit[:len(u_key)]`` has also it's problems, for example, if the unit is 'v', when you use that expression the result would be that unit = 'v' again rendering useless the function more commonly. Maybe we can use this option but then we need to restructure the function. We would need to add an if for the cases where the unit does not have prefixes

Reply:
@vinferrer is it possible that @rmarkello was suggesting to do `unit = unit[-len(u_key):]`?
Would that solve both problems?

Reply:
Sorry, similar mistake. `unit = unit[:-len(u_key)]` would solve it.
However, now that line 73 is different, line 76 is not necessary any more.
It could become (starting line 73):
`        if unit.endswith(u_key):
            new_unit = unit[-len(u_key):]
            # check that it is at the end
            unit = unit[:-len(u_key)]

            if unit != '':
                # for every prefix alias
                prefix = prefix_aliases.get(unit, '')
                if prefix == '':
                    LGR.warning(f'The given unit prefix {unit} does not have aliases, '
                                f'passing it as is')`

Reply:
I think i got it, could you check the lastest commit @rmarkello